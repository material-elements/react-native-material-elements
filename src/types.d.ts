export * from './components/types';
export * from './types/index';
