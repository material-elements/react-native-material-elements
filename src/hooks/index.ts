export * from './useRestyle';
export * from './useThemedProps';
