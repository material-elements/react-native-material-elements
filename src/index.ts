export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
