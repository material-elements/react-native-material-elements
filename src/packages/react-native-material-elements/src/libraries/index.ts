export * from './themes';
export * from './style';
