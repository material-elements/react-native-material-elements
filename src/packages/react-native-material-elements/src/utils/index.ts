export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
