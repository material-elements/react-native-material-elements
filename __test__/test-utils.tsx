import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
