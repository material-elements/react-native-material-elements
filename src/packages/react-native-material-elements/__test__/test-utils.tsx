import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
