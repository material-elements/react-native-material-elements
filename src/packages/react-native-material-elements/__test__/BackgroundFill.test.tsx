import React from 'react';
import { BackgroundFill, Text } from '../src';
import { render } from './test-utils';

describe('BackgroundFill', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<BackgroundFill />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const { toJSON } = render(
      <BackgroundFill>
        <Text>Hi</Text>
      </BackgroundFill>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the overlay', () => {
    const { toJSON } = render(<BackgroundFill overlay />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render image', () => {
    const { toJSON } = render(<BackgroundFill image={{ source: { uri: 'https://demo.image' } }} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
