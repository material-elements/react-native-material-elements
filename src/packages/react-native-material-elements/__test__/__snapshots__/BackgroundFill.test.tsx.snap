// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BackgroundFill should render children correctly 1`] = `
<View
  style={
    [
      {
        "position": "relative",
        "width": "100%",
      },
      {},
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "zIndex": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      collapsable={false}
      style={{}}
    >
      Hi
    </Text>
  </View>
</View>
`;

exports[`BackgroundFill should render correctly 1`] = `
<View
  style={
    [
      {
        "position": "relative",
        "width": "100%",
      },
      {},
      undefined,
    ]
  }
/>
`;

exports[`BackgroundFill should render image 1`] = `
<View
  style={
    [
      {
        "position": "relative",
        "width": "100%",
      },
      {},
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://demo.image",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "height": "100%",
          "left": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`BackgroundFill should render the overlay 1`] = `
<View
  style={
    [
      {
        "position": "relative",
        "width": "100%",
      },
      {},
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(0, 0, 0, .2)",
          "bottom": 0,
          "left": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 2,
        },
        undefined,
      ]
    }
  />
</View>
`;
